src/alloc_array.rs

```diff
@@ -61,3 +61,48 @@ pub fn inplace_or_alloc_array<T, R, Consumer>(size: usize, consumer: Consumer) -
         Err(consumer) => alloc_array(size, consumer),
     }
 }
+
+/// `inplace_or_alloc_from_iter` is helper function used to easy trying to place data from `Iterator`.
+///
+/// It tries to get upper bound of `size_hint` of iterator and forward it to `inplace_or_alloc_array` function.
+/// It there is not upper bound hint from iterator, then it just `collect` your data into `Vec`.
+///
+/// If iterator contains more data that `size_hint` said
```

Comment:
Sounds logical. Okay.

Reply:
Also, if we do not need it then other comments are don't need to be resolved. Because all this about possibility of iterator to return more than it `hint` to us.

Reply:
I've never seen an iterator which returns too small upper bound, but handling this case doesn't seem too hard, so why not?

Reply:
I agree. Also, because `new_from_iter` is public, it can be used separately. And, in that case, it will be good to handle this case.